array.go

```diff
@@ -118,6 +118,7 @@ func WithStartTime(start time.Time) ArrayOpenOption {
 func WithEndTimestamp(endTimestamp uint64) ArrayOpenOption {
 	return func(tdbArray *Array) error {
 		ret := C.tiledb_array_set_open_timestamp_end(tdbArray.context.tiledbContext, tdbArray.tiledbArray, C.uint64_t(endTimestamp))
+		runtime.KeepAlive(tdbArray)
```

Comment:
When in the future we support Go 1.24's [`AddCleanup`](https://pkg.go.dev/runtime@master#AddCleanup), we would have to explicitly keep the contexts alive, because the finalizers do not run in order.

Let's do it now, before marking the PR as ready for review.

Reply:
On second thought, I have misread documentation. `runtime.KeepAlive` does apply recursively (obviously), and as long as an object is kept alive, its finalizers will not run. This is what we want, and the order finalizers execute does not matter here.